src/main/scala/is/hail/annotations/Memory.scala

```diff
@@ -0,0 +1,67 @@
+package is.hail.annotations
+
+import sun.misc.Unsafe
+
+object Memory {
+  private var unsafe = {
+    val unsafeField = classOf[Unsafe].getDeclaredField("theUnsafe")
+    unsafeField.setAccessible(true)
+    unsafeField.get(null).asInstanceOf[Unsafe]
+  }
+
+  def loadBoolean(addr: Long): Boolean = unsafe.getByte(addr) != 0
+
+  def loadByte(addr: Long): Byte = unsafe.getByte(addr)
+
+  def loadShort(addr: Long): Short = unsafe.getShort(addr)
+
+  def loadInt(addr: Long): Int = unsafe.getInt(addr)
+
+  def loadLong(addr: Long): Long = unsafe.getLong(addr)
+
+  def loadFloat(addr: Long): Float = unsafe.getFloat(addr)
+
+  def loadDouble(addr: Long): Double = unsafe.getDouble(addr)
+
+  def loadAddress(addr: Long): Long = unsafe.getAddress(addr)
+
+  def storeBoolean(addr: Long, b: Boolean): Unit = unsafe.putByte(addr, if (b) 1 else 0)
+
+  def storeByte(addr: Long, b: Byte): Unit = unsafe.putByte(addr, b)
+
+  def storeShort(addr: Long, s: Short): Unit = unsafe.putShort(addr, s)
+
+  def storeInt(addr: Long, i: Int): Unit = unsafe.putInt(addr, i)
+
+  def storeLong(addr: Long, l: Long): Unit = unsafe.putLong(addr, l)
+
+  def storeFloat(addr: Long, f: Float): Unit = unsafe.putFloat(addr, f)
+
+  def storeDouble(addr: Long, d: Double): Unit = unsafe.putDouble(addr, d)
+
+  def storeAddress(addr: Long, a: Long): Unit = unsafe.putAddress(addr, a)
```

Comment:
On a 32-bit machine it is a 32-bit store.  We're allocating 64-bits in memory for addresses always, but we should be using Unsafe.ADDRESS_SIZE to allocate pointers (if we want to support 32-bit machines.  Meh.)